src/tables/sortable-table.js

```diff
@@ -36,37 +36,118 @@ import classnames from 'classnames'
 /* eslint react/jsx-key: 0 */
 
 const propTypes = {
-  columns: PropTypes.arrayOf(Types.column).isRequired,
-  controlled: PropTypes.bool.isRequired,
-  data: PropTypes.arrayOf(PropTypes.object).isRequired,
-  disableSort: PropTypes.bool.isRequired,
+  children: PropTypes.node.isRequired,
+  className: PropTypes.string,
+  data: PropTypes.arrayOf(PropTypes.object),
+  initialAscending: PropTypes.bool,
+  initialColumn: PropTypes.string,
+  disableReverse: PropTypes.bool,
+  disableSort: PropTypes.bool,
+  controlled: PropTypes.bool,
+  onChange: PropTypes.func,
   rowComponent: Types.component,
   headerComponent: Types.component,
-  className: PropTypes.string,
-  ...sortablePropTypes,
 }
-
 const defaultProps = {
-  className: ''
+  className: '',
+  data: [],
+  initialAscending: true,
+  initialColumn: '',
+  disableReverse: false,
+  disableSort: false,
+  controlled: false,
+  onChange: noop,
 }
 
-function SortableTable ({
-  columns,
+function getInitialSortControls(initialColumn, columns) {
+  const defaultControls = {
+    initialSortPath: '',
+    initialSortFunc: null,
+    initialValueGetter: null,
+  }
+  if (!initialColumn) return defaultControls
+
+  const initialProps = columns.filter(col => col.name === initialColumn).pop()
+  // Exceptional situation-- an initial column was specified but no column data
+  // exists for the named column...
+  if (!initialProps) throw new TableColumnError('initial column has no column definition')
```

Comment:
Sorry, I was misleading when I said throw a custom error- the standard error class is fine for this, as long as the message is helpful (which it is). Catching certain error instances is pretty rare in JS, so unless you need your error to do something really special, subclassing is not necessary.

Reply:
Subclassing, to me, really provides nothing more than documentation. Since it wasn't done elsewhere in this library, I removed the subclass.